pkg/build/api/types.go

```diff
@@ -76,8 +76,72 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64
+
+	// Info is a placeholder that can store multiple reasons why a build was
+	// triggered. It should cover for a future implementation of level triggered
+	// builds where you can multiple reasons why a build was triggered.
+	Info []*BuildTriggerInfo
+}
+
+// BuildTriggerInfo contains information about why/who/what triggered a build.
+type BuildTriggerInfo struct {
+	// Details is a placeholder for  information about who triggered a build if
+	// it's either a user, a token or something else
+	Details BuildTriggerDetails
```

Comment:
How about calling this Reason instead of Details?
where it's going to take values like:
"started manually by user FOO"  (or just "manual trigger" if we can't get user information)
"generic webhook trigger"
"github webhook trigger"
"config change trigger"
"triggered by image change for image FOO:BAR"